src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapperTest.java

```diff
@@ -1,19 +1,222 @@
+            Collections.emptyList(), null, null);
+        ZonedDateTime dateIssued = LocalDateTime.of(2010, 02, 25, 15, 41).atZone(TimestampService.UK_ZONE);
+
+        assertAll(
+            () -> assertThat(result.getIdentifier())
+                .first()
+                .satisfies(identifier -> assertAll(
+                    () -> assertThat(identifier.getValue()).isEqualTo("15/CH000042P/200010191704")
+                )),
+            () -> assertThat(result.getCode().getCoding())
+                .hasSize(1)
+                .first()
+                .satisfies(coding -> assertAll(
+                    () -> assertThat(coding.getCode()).isEqualTo("721981007"),
+                    () -> assertThat(coding.getDisplay()).isEqualTo("Diagnostic studies report"),
+                    () -> assertThat(coding.getSystem()).isEqualTo("http://snomed.info/sct")
+                )),
+            () -> assertThat(result.getStatus()).isEqualTo(DiagnosticReport.DiagnosticReportStatus.UNKNOWN),
+            () -> assertThat(result.getIssued()).isEqualTo(Date.from(dateIssued.toInstant()))
+        );
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithSubjectResource() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var patient = mock(Patient.class);
+        when(resourceFullUrlGenerator.generate(any(Patient.class))).thenReturn("patient-full-url");
+
+        final var result = mapper.mapToDiagnosticReport(message, patient, Collections.emptyList(),
+            Collections.emptyList(), null, null);
+
+        assertThat(result.getSubject())
+            .extracting(Reference::getReference)
+            .isEqualTo("patient-full-url");
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithSpecimenResources() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var specimen1 = mock(Specimen.class);
+        final var specimen2 = mock(Specimen.class);
+
+        when(resourceFullUrlGenerator.generate(nullable(Specimen.class))).thenReturn("some-specimen-id");
+        final var result = mapper.mapToDiagnosticReport(message, null, List.of(specimen1, specimen2),
+            Collections.emptyList(), null, null);
+
+        assertAll(
+            () -> assertThat(result.getSpecimen()).hasSize(2),
+            () -> assertThat(result.getSpecimen())
+                .first()
+                .extracting(Reference::getReference)
+                .isEqualTo("some-specimen-id"),
+            () -> assertThat(result.getSpecimen())
+                .element(1)
+                .extracting(Reference::getReference)
+                .isEqualTo("some-specimen-id")
+        );
```

Comment:
```suggestion
        assertThat(result.getSpecimen())
            .hasSize(2)
            .extracting(Reference::getReference)
            .containsExactly("some-specimen-id", "some-specimen-id");
```

Just a suggestion of how we can possibly simplify it